test/test_monitoring.py

```diff
@@ -0,0 +1,227 @@
+import mock
+from unittest import TestCase
+from datetime import datetime
+
+from biggerquery import monitoring
+from freezegun import freeze_time
+
+TEST_DATETIME = datetime(year=2019, month=1, day=2)
+TEST_DATETIME_RFC3339 = '2019-01-02T00:00:00Z'
+
+
+class FormatRFC3339TestCase(TestCase):
+
+    def test_should_format_datetime(self):
+        # expect
+        self.assertEqual(
+            monitoring.format_rfc3339(TEST_DATETIME),
+            TEST_DATETIME_RFC3339)
+
+
+class GetStartTimeTestCase(TestCase):
+
+    @freeze_time(TEST_DATETIME)
+    def test_should_return_utc_now_minus_5_minutes(self):
+        # expect
+        self.assertEqual(
+            monitoring.get_start_time(),
+            '2019-01-01T23:55:00Z')
+
+
+class GetNowRFC3339(TestCase):
+
+    @freeze_time(TEST_DATETIME)
+    def test_should_return_now_as_rfc3339(self):
+        # expect
+        self.assertEqual(
+            monitoring.get_now_rfc3339(),
+            TEST_DATETIME_RFC3339)
+
+
+class MetricExistsTestCase(TestCase):
+
+    @mock.patch('biggerquery.monitoring.api_list_metrics')
+    def test_should_check_if_metric_exists(self, api_list_metrics_mock: mock.Mock):
+        # given
+        api_list_metrics_mock.return_value = {'metricDescriptors': 'something'}
+
+        # expect
+        self.assertTrue(monitoring.metric_exists('client', 'project_resource', 'metric_type'))
+        api_list_metrics_mock.assert_has_calls([mock.call('client', 'project_resource', 'metric_type')])
+
+        # given
+        api_list_metrics_mock.return_value = {}
+
+        # expect
+        self.assertFalse(monitoring.metric_exists('client', 'project_resource', 'metric_type'))
+
+
+class WaitForMetricTestCase(TestCase):
+
+    @mock.patch('biggerquery.monitoring.metric_exists')
+    def test_should_wait_until_metric_exists(self, metric_exists_mock):
+        # given
+        metric_exists_mock.return_value = True
+
+        # expect
+        self.assertTrue(monitoring.wait_for_metric('client', 'project_resource', 'metric_type'))
+
+    @mock.patch('biggerquery.monitoring.metric_exists')
+    def test_should_raise_exception_when_no_retries_left(self, metric_exists_mock):
+        # given
+        metric_exists_mock.return_value = False
+
+        # expect
+        with self.assertRaises(monitoring.MetricError):
+            monitoring.wait_for_metric('client', 'project_resource', 'metric_type')
+
+
+class CreateTimeseriesData(TestCase):
```

Comment:
[Not important] Why you need this test ? If it properly builds data then `increment_counter()` function will properly work. I mean if we should test bigger parts of code as a whole except every small part independently ;)

Reply:
That's unit testing. If you assert created timeseries inside increment_counter test, you test 2 things in the same test(that's not unit testing, you check if increment_counter works assuming that create timeseries should work). Whether I should write the integration test is a different question :)

Reply:
Ok, that's the matter of approach. Sometimes it's better just to test bigger parts which you treat like a black box even in unit tests. Nevermind.